src/main/java/com/undertheriver/sgsg/foler/domain/Folder.java

```diff
@@ -55,6 +55,10 @@ public Folder(String title, FolderColor color, User user) {
 	public void update(FolderDto.UpdateFolderReq dto) {
 		this.title = dto.getTitle();
 	}
+
+	public void addMemo(Memo memo) {
+		this.memos.add(memo);
+	}
```

Comment:
```suggestion
	public void addMemo(Memo memo) {
		if (Objects.nonNull(memo.getFolder())) {
			memo.getFolder()
				.getMemos()
				.remove(memo);
		}
		memo.changeFolder(this);
		memos.add(memo);
	}
```
JPA관점이 아닌, 객체관점에서 memo객체는 folder가 등록되어있지 않는데요. 
memo에도 folder를 등록해주시길 바랍니다!
또 memo에 folder가 등록되어있으면 해당 폴더를 지우고, 현재 폴더에 더해주는게 좋겠네요(이미 등록되어있는 폴더는 메모를 들고 있을 수 있어요.)

추가로 remove(memo)가 잘 동작하지않을 수 있으니 id기준으로 equals, hashcode를 오버라이딩하는게 좋겠네요.

Reply:
```java
public void addMemo(Memo memo) {
		if (!this.memos.contains(memo)) {
			this.memos.add(memo);
		}
		memo.changeFolder(this);
	}
```
저도 양방향관계를 잘 안맺어봐서 잘몰랐네요ㄷ
이렇게하는게 더 심플하겠네요

Reply:
List 타입을 Set으로 변경하는 방식으로 해결해봤습니다!

Set 타입에서  add가 호출되었을 때 내부적으로 equals, hashcode가 호출된다고 하네요.
그래서 실제로 add 후 업데이트  된 내용이 잘 반영이 되는지 테스트 코드도 추가해봤습니다.

리뷰 부탁드립니다

Reply:
엔티티가 가지고 있는 List 를 Set 타입으로  변경해서 해결했습니다!

어떤 차이가 있는지 검색을 해봤는데 저희 서비스에서는  잃는 것보다 얻는 것이 훨씬 많은 것 같아서 도입했습니다.
테스트 코드도 작성해봤습니다  !  보시고 의견 부탁드립니당

잃는 것에 대한 링크: https://stackoverflow.com/questions/6562673/onetomany-list-vs-set-difference
저희 서비스와는 크게 관계 없어 보이긴 하는데 창천향로님 블로그에도 포스팅이 되어 있더라구요: https://jojoldu.tistory.com/165

Reply:
넵 확인했습니다.
중복제어도 equals, hashcode로 제어할텐데 id 기준으로 안전하게 오버라이딩 부탁드립니다!

Reply:
오버라이딩 완료 했습니다~